File: WorkcraftCore/src/main/java/org/workcraft/gui/propertyeditor/choice/ComboboxItemWrapper.java
package org.workcraft.gui.propertyeditor.choice;

import org.workcraft.util.Pair;

class ComboboxItemWrapper {
	private final Object value;
	private final String text;
	public ComboboxItemWrapper(Pair<String,? extends Object> pair) {
		this(pair.getFirst(), pair.getSecond());
	}
	public ComboboxItemWrapper(String text, Object value) {
		this.value = value;
		this.text = text;
	}
	public Object getValue() {
		return value;
	}
	@Override
	public String toString() {
		return text;
	}
}

File: Gui/src/main/scala/org/workcraft/gui/propertyeditor/choice/ComboboxItemWrapper.scala
package org.workcraft.gui.propertyeditor.choice

import org.workcraft.util.Pair


class ComboboxItemWrapper(pair : (String, Any)) {

  def getValue:Any = {
    return value
  }

  override  def toString():String = {
    return text
  }
  private def value:Any = pair._2
  private val text:String = pair._1
}
